### PR TITLE
Adding manufacturing data to the AWLAN_Node entry.

### PR DESCRIPTION
**Pull request by netexp-ravi**
Tested on ECW5211, EA8300, and EC420.

Signed-off-by: ravi vaishnav <ravi.vaishnav@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
++          }
++        },
++        "certification_region": {
++          "type": {
++            "key": {
++              "type": "string"
++            }
++          }
++        },
++        "id": {
```
Comment (mikehansen1970) [inline feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:63]: Just checking, this should replace the existing "id": column, correct?
i.e. you just removed the "min": 0, "max": 1

Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:63]: Thanks for catching this. "id" was already defined, and I happen to redefine it without min/max. I did not see any failure/complaints by the build system though. I will remove this redefinition I made.

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data [hunk 2]
```diff
@@ -2,22 +2,85 @@
...
 edgecore,ecw5410)
 	MODEL=$(cat /tmp/sysinfo/board_name | sed "s/edgecore,//" | tr [a-z] [A-Z])
-	SERIAL=$(cat /sys/class/net/eth0/address | tr -d :)
 	PLATFORM=$(cat /tmp/sysinfo/model)
+	SERIAL=$(cat /dev/mtd5 | grep serial_number | cut -d "=" -f2)
+	SKU=$(cat /dev/mtd5 | grep sku | cut -d "=" -f2)
+	CERT_REGION=$(cat /dev/mtd5 | grep certification_region | cut -d "=" -f2)
+	ID=$(cat /dev/mtd5 | grep mac_address | cut -d "=" -f2)
+	MANUF_DATE=$(cat /dev/mtd5 | grep manufacturer_date | cut -d "=" -f2)
+	MANUF_NAME="Edge-Core"
```
Comment (RickSommerville) [inline feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:29]: Based on the packaging and corporate website, they don't capitalize the "c" in "core" and they do capitalize the "E.  So "Edge-corE".  Discussed with Marcel and he agrees.

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data [hunk 3]
```diff
@@ -2,22 +2,85 @@
...
+	MODEL_REV=$(cat /dev/mtd12 | grep model_revision | cut -d "=" -f2)
+	REF_DESIGN=$(cat /dev/mtd12 | grep reference_design | cut -d "=" -f2)
+	MANUF_NAME="CIG"
+	MANUF_ABBR="CIG"
 	;;
 linksys,ea8300)
 	MODEL=$(grep modelNumber= /dev/mtd9 | tr -d '\r\n' | sed "s/modelNumber=//")
 	SERIAL=$(grep serial_number= /dev/mtd9 | tr -d '\r\n' | sed "s/serial_number=//")
-	PLATFORM=OPENWRT_EA8300
+	PLATFORM="OPENWRT_EA8300"
```
Comment (RickSommerville) [inline feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:55]: I am not sure we should change anything for Linksys (unless we maybe stub in values).  They are not a partner in the project and we have no expectation they will provide this data.

Actually I see the labels are different.  Does this mean they are providing this data in their default design with own labels?

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data [hunk 4]
```diff
@@ -28,9 +91,32 @@ esac
...
+uci set system.tip.revision="${MODEL_REV}"
+uci set system.tip.model_description="${MODEL_DESCR}"
+uci set system.tip.manufacturer_name="${MANUF_NAME}"
+uci set system.tip.manufacturer_date="${MANUF_DATE}"
+uci set system.tip.manufacturer_url="${MANUF_URL}"
+uci set system.tip.reference_design="${REF_DESIGN}"
+uci set system.tip.certification_region="${CERT_REGION}"
+uci set system.tip.id="${ID}"
+uci set system.tip.manuf_abbr="${MANUF_ABBR}"
+uci set system.tip.qr_code="${QR_CODE}"
```
Comment (RickSommerville) [inline feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:120]: Does this QR code get used?  There is no target function to retrieve it and it is constructed again in fill_entity_data().

**Review by RickSommerville — COMMENTED** [review]
Looks good.  Have a few questions though.

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
++          "type": {
++            "key": {
++              "type": "string"
++            }
++          }
++        },
++        "certification_region": {
++          "type": {
++            "key": {
++              "type": "string"
```
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:59]: We can add that limitation if we are sure that the Country code is used for the Certification region.

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
++ *
++ * This function provides a null terminated byte string containing the
++ * manufacturer name abbreviation. This parameter is required to fill the
++ * QR code entry which is a part of AWLAN_Node table.
++ *
++ * @param buff   pointer to a string buffer
++ * @param buffsz size of string buffer
++ * @return true on success
++ */
++bool target_manuf_abbr_get(void *buff, size_t buffsz);
```
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:144]: I will correct this comment. Thanks, Rick.

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 3 above)
Comment (netexp-ravi) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:55]: Mike was testing his changes against EA8300, hence I had to make these changes on the AP side. If the AP do not populate this data, then the AWLAN_Node will result in "unknown" for all the newly added parameters.

Yes, they seemed to provide most of the data except for "reference_design" and "sku_number". Only these fields go as "unknown" in the final output of AWLAN_Node.

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
++        STRSCPY(s_awlan_node->id, buff);
++        s_awlan_node->id_exists = true;
++    }
++
++    SCHEMA_KEY_VAL_APPEND(s_awlan_node->qr_code, "DT", "AP");
++    SCHEMA_KEY_VAL_APPEND(s_awlan_node->qr_code, "DM", s_awlan_node->id);
++    SCHEMA_KEY_VAL_APPEND(s_awlan_node->qr_code, "SN", s_awlan_node->serial_number);
++    SCHEMA_KEY_VAL_APPEND(s_awlan_node->qr_code, "MN", s_awlan_node->model);
++    SCHEMA_KEY_VAL_APPEND(s_awlan_node->qr_code, "HW", s_awlan_node->revision);
++    if (true == target_manuf_abbr_get(buff, sizeof(buff)))
```
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:249]: Looks like it is re-ordered before printing in the AWLAN_Node output. I will anyways update the order per the specification.

qr_code              | ["map",[["DM","C0:C9:E3:95:A7:66"],["DT","AP"],["HW","02"],["MN",           |
                     : "EC420-G1"],["SN","320B0A6000011"],["VN","PW"]]]

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 2 above)
Comment (netexp-ravi) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:29]: I see below from the flash of one of the ECW5211 I received recently. As we are hard-coding the name string. We can go ahead with "Edge-corE". The first and last letters are capitalized. Correct? Let me know.

root@OpenAp-28e498:~# cat /dev/mtd5 | grep manuf
manufacturer_date=11-13-2020
manufacturer_name=Edge-Core
manufacturer_url=www.edge-core.com

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 4 above)
Comment (netexp-ravi) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:120]: Thanks, Rick. I will remove this redundancy.
This got leftover when I moved the re-construction of the QR code from 99-tip-data to fill_entity_data().

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch (hunk 5 above)
[the remark this answers is not in the document]
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:59]: I think it is but leave it for now.

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch (hunk 6 above)
[the remark this answers is not in the document]
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch:144]: Thx

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 3 above)
Comment (RickSommerville) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:55]: OK.

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 4 above)
Comment (RickSommerville) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:120]: Thx

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 2 above)
Comment (RickSommerville) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:29]: I see in the web-page that they are supposed to provide this value and that the expected name is listed.  Therefore, I would not hard-code these values.  I would read the value provided, and use the default in the table if it is not found.

Comment (netexp-ravi) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data:29]: Makes sense. I will that change for all the platforms then.
- Will read the "manufacturer_name" from flash and commit to UCI.
- If not found in flash, hard-code it to the string from the table in https://telecominfraproject.atlassian.net/wiki/spaces/WIFI/pages/353337353/AP+Manufacturing+Data

Should we update the table in the above spec for Edge-Core to Edge-corE ??

Comment (netexp-ravi) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data]: Updated the changes to read and present the flash content for "manufacturer name", and use the table default if not present.

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch (hunk 6 above)
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch]: Done

File: feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch (hunk 5 above)
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/patches/32-add-new-manufacturing-data.patch]: Left it as-is for now.

File: feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data (hunk 2 above)
Comment (RickSommerville) [reply feeds/wlan-ap/wlan-ap-config/files/etc/uci-defaults/99-tip-data]: Leave Edge-Core as-is; they have had time to review the Confluence page and change if they wanted.

**Comment by RickSommerville** [comment]
Merged a973732, thank you.

**Pull request closed without merging**